File: lunatica-be/app/src/main/java/uk/matvey/lunatica/account/Account.java
package uk.matvey.lunatica.account;

import java.util.Optional;
import java.util.UUID;

public class Account {

    public final UUID id;
    public final String email;
    public final Optional<Long> tgChatId;

    public Account(UUID id, String email, Optional<Long> tgChatId) {
        this.id = id;
        this.email = email;
        this.tgChatId = tgChatId;
    }
}

File: lunatica-be/app/src/main/kotlin/uk/matvey/lunatica/account/Account.kt
package uk.matvey.lunatica.account

import java.time.Instant
import java.util.UUID
import java.util.UUID.randomUUID

data class Account(
    val id: UUID,
    val email: String,
    val passHash: String,
    val tgChatId: Long?,
    val createdAt: Instant,
    val updatedAt: Instant
) {
    companion object {
        fun account(email: String, passHash: String): Account {
            val now = Instant.now()
            return Account(
                randomUUID(),
                email,
                passHash,
                null,
                now,
                now
            )
        }
    }
}
